gems/smithy/lib/smithy/anvil/endpoint_parameter.rb

```diff
@@ -0,0 +1,134 @@
+# frozen_string_literal: true
+module Smithy
+  module Anvil
+    # @api private
+    class EndpointParameter
```

Comment:
Would it be weird to have a directory that is dedicated to helpers that is shared between several views?

I would say some views are not quite `client` specific since we forges some `client` views for types.
Is it necessary to have namespaces specific to whether something is client or server? 

Thinking of ways to flatten things here.